### PR TITLE
DummyMultiplex

### PR DESCRIPTION
**Pull request by annacanal**
- [x] DummyMultiplex.m docummented and updated
- [x] test_DummyMultiplex.m + nodeattack
- [x] test_DummyMultiplex.m + edgeattack

### CODE REVIEW

File: braph2/graph/graphs/test_DummyMultiplex.m [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        M(sub2ind(size(M), nodes2, nodes1)) = 0;
+    end
+    A(layer, layer) = {M};
+end    
+
+eg = g.edgeattack(g, nodes1, nodes2);
+
+assert(isequal(eg.getA(), A), ...
+    [BRAPH2.STR ':DummyMultiplex:' BRAPH2.BUG_ERR], ...
+    'Graph.edgeattack() is not working for non single layer graphs')
```
Comment (giovannivolpe) [inline braph2/graph/graphs/test_DummyMultiplex.m:66]: I haven't checked this part, but it gives an error in test_DummyMultiplex/Test3_EdgeAttack

File: braph2/graph/graphs/DummyMultiplex.m [hunk 2]
```diff
@@ -0,0 +1,174 @@
+classdef DummyMultiplex < Graph
+    % DummyMultiplex < Graph: A Multiplex graph
+    % DummyMultiplex represents a multiplex graph formed by pairs of
+    % GraphWU and GraphBD graphs.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:4]: don't call them GraphBU and GraphBDm but use the extended form

File: braph2/graph/graphs/DummyMultiplex.m [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    % It accepts even number of layers. The first half of layers are
+    % weighted undirected graphs and the second half binary directed. If
+    % the number of layers is not given, all layers will be weighted directed.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:7]: Clarify also the connections between layers and the fact that only even numbers of layers are acceptable

File: braph2/graph/graphs/DummyMultiplex.m [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    %   getDescription  - return the description of DummyMultiplex.
+    %   getGraphType    - returns the graph type.
+    %   getConnectivityType     - returns if the graph is binary or weighted.
+    %   getDirectionalityType   - returns if graph is directed or undirected.
+    %   getSelfConnectivityType     - returns if graph is self-connected or not self-connected.
+    %   getNegativityType   - returns if graph is negative or non-negative
+    %   getCompatibleMeasureList - returns a list with compatible measures.
+    %   getCompatibleMeasureNumber - returns the number of compatible measures.
+    %
+    % See also Graph, DummyMultigraph, DummyMultilayer, DummyOrderedMultigraph, DummyOrderedMultilayer.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:24]: missing DummyGraph

File: braph2/graph/graphs/DummyMultiplex.m [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            % It calls the superclass constructor GRAPH.
+            %
+            % DUMMYMULTIPLEX(A, PROPERTY1, VALUE1, PROPERTY2, VALUE2, ...) creates
+            % a DUMMYMULTIPLEX class with supra-adjacency matrix A and it passes the
+            % properties and values to the superclass as VARARGIN.
+            % This function is the constructor, it initializes the class 
+            % with the 2D-cell array of adjacency matrices A.
+            % It calls the superclass constructor GRAPH.
+            %
+            % See also Graph, DummyMultigraph, DummyMultilayer, DummyOrderedMultigraph, DummyOrderedMultilayer.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:40]: missing DummyGraph

File: braph2/graph/graphs/DummyMultiplex.m [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+            graph_type = Graph.MULTIPLEX;
+        end
+        function connectivity_type = getConnectivityType(varargin)
+            % GETCONNECTIVITYTYPE returns the connectivity type of the graph
+            %
+            % CONNECTIVITY_TYPE = GETCONNECTIVITYTYPE() returns WEIGHTED for DUMMYMULTIPLEX.
+            %
+            % CONNECTIVITY_TYPE = GETCONNECTIVITYTYPE(LAYERNUMBER) returns BINARY 
+            % for the first half of graphs and WEIGHTED for the second half for DUMMYMULTIPLEX.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:105]: Explain the possible formats of the output: single number or matrix.

Comment (giovannivolpe) [reply braph2/graph/graphs/DummyMultiplex.m:105]: also in the other methods below

File: braph2/graph/graphs/DummyMultiplex.m [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+            %
+            % DIRECTIONALITY_TYPE = GETDIRECTIONALITYTYPE(LAYERNUMBER) returns UNDIRECTED 
+            % for the first half of graphs and DIRECTED for the second half for DUMMYMULTIPLEX.
+            %
+            % See also getConnectivityType(), getGraphType(), getNegativityType() and getSelfConnectivityType().
+                       
+            if isempty(varargin)
+                directionality_type = Graph.DIRECTED;
+            else
+                layernumber = varargin{1};
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:136]: Add empty lines as in method above

File: braph2/graph/graphs/DummyMultiplex.m [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+            %
+            % SELFCONNECTIVITY_TYPE = GETSELFCONNECTIVITYTYPE() returns SELFCONNECTED for DUMMYMULTIPLEX.
+            %
+            % See also getConnectivityType(), getDirectionalityType(), getGraphType() and getNegativityType().
+                      
+            if isempty(varargin)
+                selfconnectivity_type = Graph.SELFCONNECTED;
+            else
+                layernumber = varargin{1};
+                selfconnectivity_type = Graph.SELFCONNECTED * ones(layernumber);
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:156]: add assert and spaces

File: braph2/graph/graphs/DummyMultiplex.m [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            %
+            % NEGATIVITY_TYPE  = GETNEGATIVITYTYPE() returns NONNEGATIVE for DUMMYMULTIPLEX.
+            %
+            % See also getConnectivityType(), getDirectionalityType(), getGraphType() and getSelfConnectivityType().
+            
+            if isempty(varargin)
+                negativity_type = Graph.NONNEGATIVE;
+            else
+                layernumber = varargin{1};
+                negativity_type =  Graph.NONNEGATIVE * ones(layernumber);
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:170]: add assert and spaces

File: braph2/graph/graphs/test_DummyMultiplex.m (hunk 1 above)
Comment (annacanal) [reply braph2/graph/graphs/test_DummyMultiplex.m:66]: Yes. it doesnt work. I think the edge attack function in Graph needs to be modified

File: braph2/graph/graphs/DummyMultiplex.m (hunk 6 above)
Comment (annacanal) [reply braph2/graph/graphs/DummyMultiplex.m]: done

File: braph2/graph/graphs/DummyMultiplex.m [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+            % CONNECTIVITY_TYPE = GETCONNECTIVITYTYPE(LAYERNUMBER) returns a 
+            % matrix with BINARY for the off-diagonal elements and the first 
+            % half of the diagonal; and WEIGHTED for the second half.
+            % For example, for the default matrix it returns:
+            % CONNECTIVITY_TYPE = BINARY BINARY BINARY   BINARY
+            %                     BINARY BINARY BINARY   BINARY
+            %                     BINARY BINARY WEIGHTED BINARY
+            %                     BINARY BINARY BINARY   WEIGHTED
+            % where BINARY = Graph.BINARY and WEIGHTED = Graph.WEIGHTED.
+            % If LAYERNUMBER is odd, it throws an error.
```
Comment (giovannivolpe) [inline braph2/graph/graphs/DummyMultiplex.m:122]: change documentation of subsequent methods along these lines

Comment (annacanal) [reply braph2/graph/graphs/DummyMultiplex.m:122]: done

**Pull request merged**